packages/fx-core/src/plugins/resource/frontend/ops/deploy.ts

```diff
@@ -51,60 +52,104 @@ export class FrontendDeployment {
     return lastDeployTime < lastBuildTime;
   }
 
-  public static async doFrontendBuild(
-    componentPath: string,
-    envs: RemoteEnvs,
-    envName: string
-  ): Promise<void> {
+  public static async doFrontendBuild(componentPath: string, envName: string): Promise<void> {
     if (!(await FrontendDeployment.needBuild(componentPath, envName))) {
       return FrontendDeployment.skipBuild();
     }
 
     const progressHandler = ProgressHelper.progressHandler;
 
+    const scripts =
+      (await runWithErrorCatchAndWrap(
+        (error) => new FileIOError(error.message),
+        async () => (await fs.readJSON(path.join(componentPath, PathInfo.NodePackageFile))).scripts
+      )) ?? [];
+
+    if (!("install:teamsfx" in scripts)) {
+      // * Track legacy projects
+      TelemetryHelper.sendGeneralEvent(TelemetryEvent.InstallScriptNotFound);
```

Comment:
How about adding one extra field into common properties instead? I encountered a same comment developing CICD plugin recently as you may remember.

Reply:
Which is better for this case, send separate event, or put it as property of existing event (maybe deploy event)?

Reply:
This property may be not fit to deploy event. Offline discussed with @eriolchan, let's keep this event.